lib/nylas/api.rb

```diff
@@ -37,6 +37,21 @@ def authenticate(name:, email_address:, provider:, settings:, reauth_account_id:
       )
     end
 
+    def authentication_url(redirect_uri:, scopes:, response_type: "code", login_hint: nil, state: nil)
+      scopes = scopes.join(",") if scopes
+
+      params = {
+        client_id: app_id,
+        redirect_uri: redirect_uri,
+        response_type: response_type,
+        scopes: scopes,
+        login_hint: login_hint
+      }
+      params[:state] = state if state
```

Comment:
Do we not want to include this key if it's nil?

Is this simpler if we just include the key `state: nil` and handle that downstream?